astropy/io/fits/convenience.py

```diff
@@ -460,19 +460,30 @@ def table_to_hdu(table):
     """
     # Avoid circular imports
     from .connect import is_column_keyword, REMOVE_KEYWORDS
+    
+    # Header to store Time related metadata
+    hdr = None
 
     # Not all tables with mixin columns are supported
     if table.has_mixin_columns:
         # Import is done here, in order to avoid it at build time as erfa is not
         # yet available then.
         from ...table.column import BaseColumn
+        from ...table import QTable, Column
```

Comment:
I think Table would work too. I'll have to look through it a bit more. QTable was rather an instinctive choice since we are dealing with mixins.